torch/_dynamo/variables/lists.py

```diff
@@ -100,6 +101,73 @@ def call_method(
 
         return super(BaseListVariable, self).call_method(tx, name, args, kwargs)
 
+    @staticmethod
+    def generic_list_compare(left, tx, op, right, **options):
+        from .builder import wrap_fx_proxy
+        from .tensor import DynamicShapeVariable
+
+        assert not (
+            left.is_python_constant() and right.is_python_constant()
+        ), "Illegal generic list compare on constant lists"
+
+        # Most list-like variables implement comparison ops the same way,
+        # so they can re-use this helper.
+        # There are quirks though, like how `tuple([2]) == torch.Size([2])`,
+        # but `tuple([2]) != list([2])`
+        if len(left.items) != len(right.items):
+            return ConstantVariable(False, **options)
+        if len(left.items) == 0:
+            return ConstantVariable(True, **options)
+
+        # Generic list comparison works by iterating over left aka self and right the compared-to list.
+        # If we hit here, their lengths are the same and they cannot be expressed as python constants.
+        # So, we iterate over the zipped list items.
+        equal = True
+        list_type = None
+        comps = []
+        for l_r in zip(left.items, right.items):
+            l = l_r[0]
+            r = l_r[1]
+            # When iterating over the zipped items, we invoke compare.
+            # .compare must return a DynamicShapeVariable, a TensorVariable,  or raise unimplemented
+            # If unimplemented is raised, we rely on checkpointing state to roll us back and not write these operations to the graph
+            comp = l.compare(tx, op, r, **options)
+            if list_type is None:
+                list_type = type(comp)
+            else:
+                assert (
+                    type(comp) == list_type
+                ), "Only single-type list comparison is supported atm"
```

Comment:
I wonder if this'll bite us in the near future, in a world where users mark certain dimensions as dynamic explicitly (so it could be common to compare two torch.Size arguments, that have a mix of `ConstantVariable` and `DynamicVariable`.

If you think it's worth not taking care of for now, that sounds fine to me.

Reply:
I think it's probably okay for now? I can relax this assertion more, let me do that - I was just trying to exclude tensors.